src/plugins/intel_npu/src/plugin/npuw/serialization.cpp

```diff
@@ -365,7 +369,18 @@ void ov::npuw::s11n::read_weightless(std::istream& stream,
             std::size_t offset = 0;
             read(stream, offset);
             ov::Tensor t(type, shape);
-            std::memcpy(t.data(), weights->get_ptr(offset), byte_size);
+
+            if (ctx.weights) {
+                std::memcpy(t.data(), ctx.weights->get_ptr(offset), byte_size);
+            } else {
+                auto it = ctx.consts_cache.find({offset, byte_size});
+                NPUW_ASSERT(it != ctx.consts_cache.end() && "Couldn't find Constant in cache!");
+                auto tensor = ov::npuw::util::tensor_from_const(it->second);
+                NPUW_ASSERT(tensor.get_byte_size() == byte_size && tensor.get_shape() == shape &&
+                            tensor.get_element_type() == type);
+                tensor.copy_to(t);
```

Comment:
what's the reason to make an explicit tensor copy if you already have the right tensor in your consts_cache?
why don't you put `tensor` to `var` directly?

Reply:
Copy is required here since we don't have a Const owning a memory during deserialization and in tensor_from_const we just construct on top of a pointer. Thus we need to explicitly do copy when reading weights so the tensor is alive in closure after the compilation